ceph_deploy/util/init_systems.py

```diff
@@ -0,0 +1,246 @@
+from ceph_deploy.lib import remoto
+
+# Provide mappings/translations for services depending
+# on Init system.  All services coming into the classes
+# here should be one of ceph-mon, ceph-osd, ceph-mds, or
+# ceph-radosgw. It's then up to the code here to Do The
+# Right Thing.
+SYSV_SERVICES = {
+    'ceph-mon': 'mon',
+    'ceph-mds': 'mds'
+}
+
+UPSTART_SERVICES = {
+    'ceph-radosgw': 'radosgw'
+}
+
+
+class InitSystem(object):
+    """
+    Base class for all Init systems
+    """
+
+    def __init__(self, remote_conn):
+        self.remote_info = remote_conn
+        self.remote_conn = remote_conn.conn
+        self.hostname = self.remote_conn.remote_module.shortname()
+
+    def _run(self, cmd, **kw):
+        if 'timeout' not in kw:
+            kw['timeout'] = 7
+        return remoto.process.run(
+            self.remote_conn,
+            cmd,
+            **kw
+        )
+
+    def start(self, service, **kw):
+        """Start service on remote node"""
+        raise NotImplementedError()
+
+    def stop(self, service, **kw):
+        """Stop service on remote node"""
+        raise NotImplementedError()
+
+    def enable(self, service, **kw):
+        """Enable service on remote node"""
+        raise NotImplementedError()
+
+    def disable(self, service, **kw):
+        """Disable service on remote node"""
+        raise NotImplementedError()
+
+    def status(self, service, **kw):
```

Comment:
I feel that status method here maybe redundant. I wonder if it should be replaced with some code to establish if the service is running which returns a boolean true or false.

Admittedly in master so far we do not have a use case yet for status, but I am using it in the SUSE rgw code, I use similar code to this to establishing if apache is running.

For a tested implementation for systemd:

https://github.com/osynge/ceph-deploy/commit/09dc6213e645ee2fe7bbc16d82b21cc350f2ceb3